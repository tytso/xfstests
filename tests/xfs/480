#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (c) 2018 Oracle, Inc.  All rights reserved.
#
# FS QA Test No. 480
#
# Populate a XFS filesystem and fuzz every external attr block field.
# Do not fix the filesystem, to test metadata verifiers.

. ./common/preamble
_begin_fstest dangerous_fuzzers dangerous_norepair

_register_cleanup "_cleanup" BUS

# Import common functions.
. ./common/filter
. ./common/populate
. ./common/fuzzy

# real QA test starts here
_supported_fs xfs
_require_scratch_xfs_fuzz_fields
_disable_dmesg_check

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

echo "Find external attr block"
_scratch_mount
inum=$(stat -c '%i' $SCRATCH_MNT/ATTR.FMT_EXTENTS_REMOTE3K)
_scratch_unmount

echo "Fuzz external attr block"
_scratch_xfs_fuzz_metadata '' 'none'  "inode ${inum}" "ablock 1" >> $seqres.full
echo "Done fuzzing external attr block"

# success, all done
status=0
exit
