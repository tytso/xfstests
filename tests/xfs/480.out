QA output created by 480
Format and populate
Find external attr block
Fuzz external attr block
online health check (0) with hdr.magic = zeroes.
online health check (0) with hdr.magic = ones.
online health check (0) with hdr.magic = firstbit.
online health check (0) with hdr.magic = middlebit.
online health check (0) with hdr.magic = lastbit.
online health check (0) with hdr.magic = add.
online health check (0) with hdr.magic = sub.
online health check (0) with hdr.offset = ones.
online health check (0) with hdr.offset = firstbit.
online health check (0) with hdr.offset = middlebit.
online health check (0) with hdr.offset = lastbit.
online health check (0) with hdr.offset = add.
online health check (0) with hdr.offset = sub.
online health check (0) with hdr.bytes = zeroes.
online health check (0) with hdr.bytes = ones.
online health check (0) with hdr.bytes = firstbit.
online health check (0) with hdr.bytes = middlebit.
online health check (0) with hdr.bytes = lastbit.
online health check (0) with hdr.bytes = add.
online health check (0) with hdr.bytes = sub.
online health check (0) with hdr.crc = zeroes.
online health check (0) with hdr.crc = ones.
online health check (0) with hdr.crc = firstbit.
online health check (0) with hdr.crc = middlebit.
online health check (0) with hdr.crc = lastbit.
online health check (0) with hdr.crc = add.
online health check (0) with hdr.crc = sub.
online health check (0) with hdr.uuid = zeroes.
online health check (0) with hdr.uuid = ones.
online health check (0) with hdr.uuid = firstbit.
online health check (0) with hdr.uuid = middlebit.
online health check (0) with hdr.uuid = lastbit.
online health check (0) with hdr.owner = zeroes.
online health check (0) with hdr.owner = ones.
online health check (0) with hdr.owner = firstbit.
online health check (0) with hdr.owner = middlebit.
online health check (0) with hdr.owner = lastbit.
online health check (0) with hdr.owner = add.
online health check (0) with hdr.owner = sub.
online health check (0) with hdr.bno = zeroes.
online health check (0) with hdr.bno = ones.
online health check (0) with hdr.bno = firstbit.
online health check (0) with hdr.bno = middlebit.
online health check (0) with hdr.bno = lastbit.
online health check (0) with hdr.bno = add.
online health check (0) with hdr.bno = sub.
online health check (0) with data = zeroes.
online health check (0) with data = ones.
online health check (0) with data = firstbit.
online health check (0) with data = middlebit.
online health check (0) with data = lastbit.
Done fuzzing external attr block
